tensorflow/python/keras/layers/wrappers.py

```diff
@@ -148,56 +150,73 @@ def _get_shape_tuple(self, init_tuple, tensor, start_idx, int_shape=None):
         of the output shape
 
     Returns:
-      The new int_shape with the first part from init_tuple
-      and the last part from either `int_shape` (if provided)
-      or `tensor.shape`, where every `None` is replaced by
-      the corresponding dimension from `tf.shape(tensor)`.
+      A tensor with the new int_shape with the first part from
```

Comment:
Any reason to convert the output to a tensor?

Reply:
Yes, because the return values of the `_get_shape_tuple` function will be used within `nest.map_structure` calls and if they were a list or tuple the `nest.map_structure` call would apply the specified function to every element of the list. However, the provided functions should be applied to every shape (as a whole) and not to every element of the shape. Therefore, the output is converted to a tensor.